hibernate-core/src/test/java/org/hibernate/orm/test/temporal/MySQLTimestampFspFunctionTest.java

```diff
@@ -51,8 +52,27 @@ public void testTimeStampFunctions(SessionFactoryScope scope) {
 					final Timestamp now = (Timestamp) oArray[0];
 					assertEquals( now, oArray[1] );
 					assertEquals( now, oArray[2] );
+					assertTrue( now.compareTo( (Timestamp) oArray[3] ) <= 0 );
+				}
+		);
+	}
+	@Test
+	public void testTimeFunctions(SessionFactoryScope scope) {
+		// the returned TIME values for now(), current_timestamp(), localtime(), and
+		// localtimestamp() should be the same.
+		// sysdate() is the time at which the function itself is executed, so the
+		// value returned for sysdate() should be different.
+		scope.inSession(
+				s -> {
+					Query q = s.createQuery(
+							"select CAST(now() AS TIME), CAST(current_timestamp() AS TIME), CAST(localtime() AS TIME), CAST(localtimestamp() AS TIME), CAST(sysdate() AS TIME)"
+					);
+					Object[] oArray = (Object[]) q.uniqueResult();
+					final Time now = (Time) oArray[0];
+					assertEquals( now, oArray[1] );
+					assertEquals( now, oArray[2] );
 					assertEquals( now, oArray[3] );
-					assertTrue( now.compareTo( (Timestamp) oArray[4] ) <= 0 );
+					assertTrue( now.compareTo( (Time) oArray[4] ) <= 0 );
```

Comment:
I think the problem you're encountering is that `MySQLDialect` wrongly uses the `TIMESTAMP` type as return type for `localtime`, though I'm wondering why this is a problem. Did MariaDB or the driver get stricter in some sense?

Reply:
with the 3.x version of the connector, yes, it follows JDBC recommendation, and for TIME, it will return value of 1970-01-01 + time, not current date + time. That's why this change

Reply:
Thanks for all your work so far. I'll experiment with a [fix](https://github.com/hibernate/hibernate-orm/pull/9472) for this `localtime` issue before merging this, because I don't think the changes in this file should be necessary.

The idea is, that you rebase after I fixed problems with `localtime` and potentially just revert changes in this file. Then we can merge your PR.

Reply:
Ok, I merged a fix for this now. Please rebase again.